src/main/java/org/terasology/launcher/packages/CustomRepositoryHandler.java

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright 2019 MovingBlocks
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.terasology.launcher.packages;
+
+import java.util.List;
+
+/**
+ * Handles fetching of packages from any custom repository.
+ * Should be used to handle any third party repository.
+ */
+class CustomRepositoryHandler implements RepositoryHandler {
+    @Override
+    public List<Package> getPackageList(PackageDatabase.Repository source) {
+        // TODO: Implement this
```

Comment:
In cases like this a logging statement would help to spot non-implemented methods. If you are sure that this is not called yet, you could even throw an [`UnsupportedOperationException`](https://docs.oracle.com/javase/8/docs/api/java/lang/UnsupportedOperationException.html) in these cases.